Fixes #188: Add Marker Widget section to GUI Widget Guide

danielkaijzer opened the pull request:
This change addresses [Issue #188](https://github.com/OpenBCI/Documentation/issues/188).

## Changes:
- Added a new Marker Widget section to the GUI Widget Guide.
- Explained the three ways to insert markers: UI buttons, keyboard shortcuts, and UDP input. 
- Documented how to configure UDP marker input, including IP/Port settings.
- Provided a link to the [example Python script](https://github.com/OpenBCI/OpenBCI_GUI/blob/db1cbc580980f725c85f6d46ec98e2f7cefb9851/Networking-Test-Kit/UDP/udp_send_marker.py) in the Networking Test Kit.

## Notes
- I didn’t update the OpenBCI GUI Networking Guide since I don’t have access to the Google Docs file.
- Can add images/GIFs to make this information even more clear
- This page (GUI Widget Guide) uses the [Github Releases page](https://github.com/OpenBCI/OpenBCI_GUI/releases) for the GUI download link, whereas the OpenBCI GUI page uses the [website download link](https://openbci.com/downloads). Should we standardize this and use the website link here as well?

zoeis52 commented on website/docs/Software/OpenBCISoftware/02_GUI_Widget_Guide.md:
```diff
@@ -273,6 +273,30 @@ Open Sound Control is a protocol for networking sound synthesizers, computers, a
+      * Shift + Z, Shift + X, Shift + C, Shift + V → Insert markers 5, 6, 7, and 8.
+3. **UDP Input:** The GUI can receive markers from an external program over a network using UDP (User Datagram Protocol). This allows markers to be inserted remotely from another application.
+
+### Using UDP to Send Markers
```
Should also probably have a section on how to receive markers

danielkaijzer replied:
Added a new section "Viewing and Accessing Markers" to explain how markers are displayed in the GUI and saved in recording files.

zoeis52 commented on website/docs/Software/OpenBCISoftware/02_GUI_Widget_Guide.md:
```diff
@@ -273,6 +273,30 @@ Open Sound Control is a protocol for networking sound synthesizers, computers, a
+1. **UI Buttons:** Click one of the marker buttons within the GUI to insert a marker manually. You can adjust the vertical scale inside the widget to increase or decrease the number of available marker buttons.
+2. **Keyboard Shortcuts:** Use the following predefined keyboard shortcuts to add markers quickly:
+      * Z, X, C, V → Insert markers 1, 2, 3, and 4.
+      * Shift + Z, Shift + X, Shift + C, Shift + V → Insert markers 5, 6, 7, and 8.
```
Good you found these. Mapping is clear, but could be nicer with a table

danielkaijzer replied:
Added a table per your recommendation. I definitely think this looks better.

zoeis52 commented on website/docs/Software/OpenBCISoftware/02_GUI_Widget_Guide.md:
```diff
@@ -273,6 +273,30 @@ Open Sound Control is a protocol for networking sound synthesizers, computers, a
+- When an external application sends a single float value, the GUI records that value as a marker.
+- Updating the IP address or port in the GUI will automatically restart the UDP listener.
+
+Here is an **[example Python script in the GUI Networking Test Kit](https://github.com/OpenBCI/OpenBCI_GUI/blob/db1cbc580980f725c85f6d46ec98e2f7cefb9851/Networking-Test-Kit/UDP/udp_send_marker.py)** that sends markers over UDP to the OpenBCI GUI.
```
Good to link this

zoeis52 reviewed (commented):
Hi Daniel, apologies for taking so long to get this review back to you.
Overall I like that this solution is short and sweet. Its good that you found the issue with the networking guide, which should probably be moved out of google docs to the documentation repo here. Just curious to not include screenshots even though you mention that it would be useful for the documentation.
Thank you again for interviewing with us!

danielkaijzer commented:
Hi Zoe,
Thank you for the feedback! 
Per your recommendation, I added:
- a new section explaining how to view and access markers
- an image showing the Marker Widget in action
- a keyboard shortcuts table for better readability

I also added info about recording file locations just in case.
